src/Particles/babylon.particleSystem.ts

```diff
@@ -555,10 +571,139 @@
             this._vertexData[offset + 11] = particle.cellIndex;
         }
 
+        // start of sub system methods
+
+        /**
+         * "Recycles" one of the particle by copying it back to the "stock" of particles and removing it from the active list.
+         * Its lifetime will start back at 0.
+         */
+        public recycleParticle: (particle: Particle) => void = (particle) => {
+            this._recycleParticleUsingSystem(this, this, particle);
+        };
+
+        private _stopSubEmitters(): void {
+            this.stockSubSystems.forEach(index => {
+                this.stockSubSystems.get(index)!.forEach(subSystem => {
+                    subSystem.stop();
+                });
+            });
+
+            this.activeSubSystems.forEach(subSystem => {
+                subSystem.stop();
+                subSystem._stoppedEmitting();
+            });
+            this.activeSubSystems = new Array<ParticleSystem>();
+        }
+
+        private _createParticle: () => Particle = () => {
+            return this._createParticleUsingSystem(this, this);
+        }
+
+        // to be overriden by subSystems
+        private _stoppedEmitting: () => void = () => {
+
+        }
+
+        private _emitFromParticle: (particle: Particle) => void = (particle) => {
+            if (!this.subEmitters || this.subEmitters.length === 0) {
+                return;
+            }
+
+            var templateIndex = Math.floor(Math.random() * this.subEmitters.length);
+            var templateIndexString =templateIndex.toString();
+
+            if (!this.stockSubSystems.contains(templateIndexString) || (this.stockSubSystems.contains(templateIndexString) && this.stockSubSystems.get(templateIndexString)!.length === 0)) {
+                // get the current generation template and clone it to subSystem
+                var subSystem = this.subEmitters[templateIndex]._cloneToSubSystem(this, particle.position);
+                subSystem._subEmitterIndex = templateIndexString;
+                this.activeSubSystems.push(subSystem);
+                subSystem.start();
+            }
+            else {
+                var stockSubSystem = this.stockSubSystems.get(templateIndexString)!.pop()!;
+                stockSubSystem.emitter = particle.position;
+                this.activeSubSystems.push(stockSubSystem);
+
+                // reset the manual emit count
+                if (this.subEmitters[templateIndex].manualEmitCount != -1)
+                    stockSubSystem.manualEmitCount = this.subEmitters[templateIndex].manualEmitCount;
+
+                stockSubSystem.start();
+            }
+        }
+
+        private _initSubSystem(rootParticleSystem: ParticleSystem): void {
+            this._rootParticleSystem = rootParticleSystem;
+
+            this._stoppedEmitting = () => {

```

Comment:
subSystem are regular system. They just need to keep a ref to the originator in order to be stopped simultaneously on demand. They do not reply on it for particle emission or recycling. Every system is completely independant